src/validate.ts

```diff
@@ -121,20 +114,25 @@ export function validate(intentRequest: object, apiResponse: object, syncData?:
    // identifies the part of the api response to validate against a schema
     const commands = apiResponse['payload']['commands'];
     const commandsLength = commands.length;
-    for (let i = 0; i < executionLength; i++) {
-      // gets the specific command
-      const commandName = execution[i]['command'];
-      if (commandName in COMMANDS) {
-        for (let j = 0; j < commandsLength; j++) {
-          const states = commands[j]['states'];
-          const validateExecTraitStates = responseValidation(states, COMMAND_STATES_EXPECT[COMMANDS[commandName]]);
-          if (validateExecTraitStates) {
-            executeErrors.push(...validateExecTraitStates);
-          } else {
-            return validateExecuteAPI;
+
+    if (syncData){
+        const syncDevices = syncData['payload']['devices'];
+        const syncDevicesLength = syncDevices.length;
+        for (let i = 0; i < syncDevicesLength; i++) {
+          const trait = syncDevices[i]['traits'];
+          // gets the specific command
+          if (trait in COMMAND_STATES_EXPECT) {
+            for (let j = 0; j < commandsLength; j++) {
+              const states = commands[j]['states'];
+              const validateExecTraitStates = responseValidation(states, COMMAND_STATES_EXPECT[trait]);
```

Comment:
This would not necessarily be a reliable mechanism, as even if a device has more than one trait, they only need to return the states for a particular command. As such, the OnOff command would not include brightness or color even if the device has those traits. So you'd get a lot of states missing even though that's okay.